arch/xtensa/src/esp32s3/esp32s3_spiflash_mtd.c

```diff
@@ -435,9 +435,10 @@ static ssize_t esp32s3_read(struct mtd_dev_s *dev, off_t offset,
   ssize_t ret;
 
 #ifdef CONFIG_ESP32S3_STORAGE_MTD_DEBUG
-  finfo("%s(%p, 0x%x, %d, %p)\n", __func__, dev, offset, nbytes, buffer);
+  finfo("%s(%p, 0x%jx, %d, %p)\n",
+        __func__, dev, (intmax_t)offset, nbytes, buffer);
 
-  finfo("spi_flash_read(0x%x, %p, %d)\n", offset, buffer, nbytes);
+  finfo("spi_flash_read(0x%jx, %p, %d)\n", (intmax_t)offset, buffer, nbytes);
```

Comment:
```suggestion
  finfo("spi_flash_read(0x%" PRIxOFF ", %p, %d)\n", (offset, buffer, nbytes);
```
and apply the suggestion to other change too.